src/serac/physics/base_physics.cpp

```diff
@@ -168,10 +187,18 @@ void BasePhysics::outputStateToDisk(std::optional<std::string> paraview_output_d
       for (const FiniteElementState* state : states_) {
         state->gridFunction();  // update grid function values
       }
+      for (const FiniteElementDual* dual : duals_) {
+        // These are really const calls, but MFEM doesn't label them as such
+        serac::FiniteElementDual* non_const_dual = const_cast<serac::FiniteElementDual*>(dual);
+        non_const_dual->space().GetRestrictionMatrix()->MultTranspose(*dual,
+                                                                      *paraview_dual_grid_functions_[dual->name()]);
+      }
```

Comment:
It seems to me that the repetition of this code (from lines 156-161) be avoided without too much effort. If I'm wrong, forget this comment, but otherwise, consider refactoring.